.github/workflows/check_code_format.yml

```diff
@@ -0,0 +1,36 @@
+---
+name: check_code_format
+
+'on':
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  check_format_code:
+    name: check format code
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install fprettify
+        run: pip install fprettify
+
+      - name: Display fprettify version
+        run: fprettify --version
+
+      - name: Format code
+        run: |
+          git clean -f -x -d
+          fprettify --indent 4 --recursive .
+
+      - name: Fail if needs reformatting
```

Comment:
```
      - name: Fail if needs reformatting      
        run: |  
          if [[ $(git status --porcelain) ]]; then            
            echo "please reformat/fprettify these files:"            
            git diff
            exit 1
          else
            echo "All files are properly formatted."
          fi
```

Reply:
It works now, cf. https://github.com/TheAlgorithms/Fortran/actions/runs/11045781263/job/30683988597